### PR TITLE
Revert workaround for mdsd bug

### PR DESCRIPTION
**Pull request by s-fairchild**
### Which issue this PR addresses:

https://issues.redhat.com/browse/ARO-1535

### What this PR does / why we need it:

Remove workaround that reset log permissions to syslog. Once Microsoft fixes race condition [in this card](https://dev.azure.com/msazure/One/_workitems/edit/12512148) this PR will be ready to merge.

This PR reverts a workaround (resetting log permissions to syslog) for a MDSD race condition.
When this problem occurs, odd and hard to track down logging behavior can occur. In this case of [this ICM](https://portal.microsofticm.com/imp/v3/incidents/details/269121287/home) log messages were sent to dgrep, but not kusto. Alerts were received that AzSecPack was not running, but dgrep showed proof it was running.

On the RP VMSS an MDSD race event was occurring. Threads were using different uids to create files/directories, resulting in MDSD losing permission to those directories. See ICM comments for detailed error message examples.

The version of MDSD affected was not found in notes or attached log mesages.

### Test plan for issue:

Tested by deploying pull request to INT. [Deployment and E2E passed](https://dev.azure.com/msazure/AzureRedHatOpenShift/_releaseProgress?_a=release-pipeline-progress&releaseId=3765)

### Is there any documentation that needs to be updated for this PR?
No

### CODE REVIEW

**Comment by s-fairchild** [comment]
@microsoft-github-policy-service agree [company="{Red Hat}"]

**Review by s-amann — APPROVED** [review]
lgtm, nice work! Let's continue to hold merging this until MSFT has their fix in place.

**Review by ross-bryan — APPROVED** [review]

**Review by ross-bryan — DISMISSED** [review]

**Comment by SudoBrendan** [comment]
Hey @s-fairchild - CI is failing due to not committing the files created after running `make generate`. Can you do so?

File: pkg/deploy/generator/scripts/gatewayVMSS.sh [hunk 1]
```diff
@@ -434,17 +434,6 @@ cat >/etc/default/vsa-nodescan-agent.config <<EOF
...
-# by the correct user as it gets created by root and may cause a race condition
-# where root owns the dir instead of syslog
-# TODO: https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/12591207
-cat >/etc/cron.d/mdsd-chown-workaround <<EOF
-SHELL=/bin/bash
-PATH=/bin
-0 * * * * root chown syslog:syslog /var/opt/microsoft/linuxmonagent/eh/EventNotice/arorplogs*
-EOF
-
-echo "enabling aro services"
```
Comment (SudoBrendan) [inline pkg/deploy/generator/scripts/gatewayVMSS.sh:447]: I feel like we'll still want these `echo` messages for the loop that comes after

File: pkg/deploy/generator/scripts/rpVMSS.sh [hunk 2]
```diff
@@ -619,17 +619,6 @@ cat >/etc/default/vsa-nodescan-agent.config <<EOF
...
-# by the correct user as it gets created by root and may cause a race condition
-# where root owns the dir instead of syslog
-# TODO: https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/12591207
-cat >/etc/cron.d/mdsd-chown-workaround <<EOF
-SHELL=/bin/bash
-PATH=/bin
-0 * * * * root chown syslog:syslog /var/opt/microsoft/linuxmonagent/eh/EventNotice/arorplogs*
-EOF
-
-echo "enabling aro services"
```
Comment (SudoBrendan) [inline pkg/deploy/generator/scripts/rpVMSS.sh:632]: same as above.

**Review by SudoBrendan — CHANGES_REQUESTED** [review]
I think we'll want to keep the log messages for other logic.

**Comment by s-fairchild** [comment]
@SudoBrendan Yes, I forgot to generate the files. I'll run `make generate` and add the enabling aro services echo back.

**Comment by s-fairchild** [comment]
[This pull request](https://dev.azure.com/msazure/One/_git/Compute-Runtime-Tux/pullrequest/8454967?path=/CMakeLists.txt) was merged fixing the MDSD bug.  

I'm trying to determine which image version will contain this fix. Once we have confirmed the MDSD image will be updated to include the fix we can merge/rollout this change.

**Comment by s-fairchild** [comment]
Fix has been merged and [can be expected in MDSD rollout version 1.28](https://dev.azure.com/msazure/One/_workitems/edit/12512148#24894147). This PR should be merged after we've updated to the 1.28 (or later) image.

**Comment by bennerv** [comment]
@s-fairchild  is this ready to go with our most recent MDSD bump?  #3106

**Comment by s-fairchild** [comment]
@bennerv Yes, it's ready with the next MDSD image update.

**Comment by s-fairchild** [comment]
[Deployment and E2E passed in INT](https://dev.azure.com/msazure/AzureRedHatOpenShift/_releaseProgress?_a=release-pipeline-progress&releaseId=3765)

**Review by s-amann — APPROVED** [review]

**Comment by hlipsig** [comment]
Dependency for this card is completed, but card is over a year old. Closing and @s-fairchild  if this is needed please re-open.

**Comment by s-fairchild** [comment]
@hlipsig Confirming that the dependency card had been completed took some time. After this I needed to test in INT. This PR is ready to merge for the next release.

**Review by cadenmarchese — APPROVED** [review]

**Pull request merged**